### PR TITLE
Tests for March Safety

### LINKED ISSUE #1: Configuring the programming environment
Setting up the basic programming environment and integrate the following tools:

- [x] C++
- [x] Style guide
- [x] Gtest
- [x] ROS
- [x] GitHub

### PR DESCRIPTION
**Pull request by TimBuckers**
## Issues

Connects https://github.com/project-march/march-iv/issues/169

## Description

March safety tested. 
- Normal behavior, cases in which the system operates withing the normal boundaries. No error should be thrown.
- Bad weather tests, tests cases that should trigger an error
- A parameterized test to prevent bugs like off-by-one error
> There are two hard things in computer science: cache invalidation, naming things, and off-by-one errors.

## Run
you can run `build.sh` 
or `catkin_make tests; rostest --text march_safety march_safety.test`

### CODE REVIEW

**Comment by TimBuckers** [comment]
I added some comments for tests of which the name is not sufficient.
I think names like this are explaining enough:
`TEST_F(TestError, exceedDefaultThresholdMultipleTimes)`

File: march_ws/src/march_safety/test/TestError.cpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+  uint32_t count;
+};
+
+class TestError : public ::testing::Test
+{
+protected:
+};
+
+TEST_F(TestError, exceedSpecificThreshold)
```
Comment (Ishadijcks) [inline march_ws/src/march_safety/test/TestError.cpp]: TestTemperatureError

File: march_ws/src/march_safety/test/TestError.cpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  EXPECT_EQ(times, errorCounter.count);
+}
+
+/**
+ * The main method which runs all the tests
+ */
+int main(int argc, char** argv)
+{
+  ROS_INFO("run main method of test");
```
Comment (Ishadijcks) [inline march_ws/src/march_safety/test/TestError.cpp]: Improve description

File: march_ws/src/march_safety/test/TestError.cpp [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  ros::init(argc, argv, "march_safety_test");
+  testing::InitGoogleTest(&argc, argv);
+  // WHEN RUNNING ALL TESTS WAITING TIME IS NOT WORKING
+  // PUBLISHERS ARE NOT REMOVED??
+  //  ::testing::GTEST_FLAG(filter) = "TestError.exceedDefaultThresholdMultipleTimes";
```
Comment (Ishadijcks) [inline march_ws/src/march_safety/test/TestError.cpp]: Remove comments

File: march_ws/src/march_safety/test/TestNoError.cpp [hunk 4]
```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 Project March.
+
+#include <march_shared_resources/TopicNames.h>
+#include "gtest/gtest.h"
+#include "ros/ros.h"
+#include "../src/TemperatureSafety.h"
```
Comment (Ishadijcks) [inline march_ws/src/march_safety/test/TestNoError.cpp]: Include directly from package?

Comment (Ishadijcks) [reply march_ws/src/march_safety/test/TestNoError.cpp]: Maybe we need to refactor the package to
```
/include/<package>*.h
/src/*.cpp
```
As seen in `march_hardware`

File: march_ws/src/march_safety/test/TestNoError.cpp [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+TEST_F(TestNoError, belowSpecificThreshold)
+{
+  ros::NodeHandle nh;
+  ros::Publisher pub_joint1 = nh.advertise<sensor_msgs::Temperature>("march/temperature/test_joint1", 0);
+  ErrorCounter errorCounter;
+  ros::Subscriber sub = nh.subscribe("march/error", 0, &ErrorCounter::cb, &errorCounter);
+
+  while (0 == pub_joint1.getNumSubscribers() || 0 == sub.getNumPublishers())
+  {
+    ros::Duration(0.1).sleep();
```
Comment (Ishadijcks) [inline march_ws/src/march_safety/test/TestNoError.cpp]: How long do you expect this to take? Milliseconds or upwards to 1 sec?

File: march_ws/src/march_safety/test/TestNoError.cpp [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  }
+  EXPECT_EQ(1, pub_joint1.getNumSubscribers());
+  EXPECT_EQ(1, sub.getNumPublishers());
+
+  sensor_msgs::Temperature msg;
+  msg.temperature = 59;
```
Comment (Ishadijcks) [inline march_ws/src/march_safety/test/TestNoError.cpp]: Maybe make these values dynamic? Get the threshold from the launchfile and subtract a value from it?

File: march_ws/src/march_safety/test/TestParameterized.cpp [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+#include <march_shared_resources/TopicNames.h>
+#include "gtest/gtest.h"
+#include "ros/ros.h"
+#include "../src/TemperatureSafety.h"
+
+/**
+ * Counter
+ */
+struct ErrorCounter
```
Comment (Ishadijcks) [inline march_ws/src/march_safety/test/TestParameterized.cpp]: Extract and include

File: march_ws/src/march_safety/test/TestNoError.cpp (hunk 5 above)
Comment (TimBuckers) [reply march_ws/src/march_safety/test/TestNoError.cpp]: upwards to 1 sec

**Pull request merged**